gramps/gen/filters/optimizer.py

```diff
@@ -18,29 +19,18 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 
+from functools import reduce
 import logging
 
 LOG = logging.getLogger(".filter.optimizer")
 
 
 def intersection(sets):
-    if sets:
-        result = sets[0]
-        for s in sets[1:]:
-            result = result.intersection(s)
-        return result
-    else:
-        return set()
+    return reduce(lambda x, y: x & y, sets, sets[0] if sets else set())
```

Comment:
if any set in handlesets is empty, we can optimize by immediately returning set()

Reply:
I generalized your basic idea: if you put the shorter sets first, then it will reduce faster.

Reply:
Nice!
Given that you now have a sorted list of sets, for one if statement, you could test if sets[0] has length zero. Whether it is worth it is a different matter!

Reply:
True, but it may be more expensive to do the test :) Anyway, I think this is clean, elegant, and not prone to errors.